x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -524,6 +524,12 @@ class AgentPolicyService {
       agentPolicy.is_protected = false;
     }
 
+    if (existingAgentPolicy.is_managed && agentPolicy?.is_protected) {
```

Comment:
Since we're only hiding the tokens instead of preventing generation, we can probably remove this check. It won't do anything now even if it is somehow enabled but in the future there might be non-endpoint agents that utilize tamper protection.

Reply:
yup ur right! i forgot to remove this

Reply:
@joeypoon 
Don't we still need an api guard though?

Reply:
after talking with @joeypoon, the explicit api guard would have been necessary if we were not going to generate uninstall tokens for managed policies, but now that we will still be generating uninstall tokens for managed policies and the UI already hides the option to turn on agent tamper protection, even if the user were to try to force tamper protection, it would be a NOOP